modules/pool/lib/pool-on-chain-data.service.ts

```diff
@@ -246,7 +252,15 @@ export class PoolOnChainDataService {
         for (const poolData of poolDataPerPool) {
             if (poolData.ignored) {
                 console.log(`Pool query return with error, skipping: ${poolData.id}`);
-                continue;
+                // Set swapEnabled to false to avoid issues with b-sdk considering these pools for swaps
+                await prisma.prismaPoolDynamicData.update({
+                    where: { id_chain: { id: poolData.id, chain: networkContext.chain } },
+                    data: {
+                        swapEnabled: false,
+                        blockNumber,
+                    },
+                });
```

Comment:
This is true if the pool is "broken". I'm thinking about scenarios where a pool is being created, e.g. a linear pool that has just been created but not yet init joined. This would set this to false. I think it will set it to true once it updates because of the init join transaction. Can you double check if the init join triggers an update on the pool?

Reply:
Not sure I understand fully. I thought the onchain data was fetched on a cron job? So then it should be updated correctly when the pool is initialised because the pool query won't throw an error.

Reply:
Only changed pools will be synced: https://github.com/beethovenxfi/beethovenx-backend/blob/swapEnabled-fix/modules/pool/lib/pool-sync.service.ts#L35-L37

But I assume an init join would also trigger a 'PoolBalanceChanged' event